Design/Instrument-2.0/Instrument_Ray_Tracer_2.md

```diff
@@ -68,6 +68,9 @@ Before rolling out any changes, it is probably a good idea to develop a set of t
 #### Changes to the Code Base
 The best way to completely move away from using `InstrumentRayTracer` (and not completely break everything) would be to start with a selection of "testing files" that currently use the `InstrumentRayTracer` e.g `Peak` and begin to change them one by one. Once all of those files are in working order it would provide enough confidence in the new implementation and all the other files can then be updated.
 
+#### Changes to Peak
+`Peak.h` has a member variable called `m_inst`. This is the kind of thing we would like to get away from using. By using `ComponentInfo` we can replace almost all the calls made using the `m_inst` variable. However, passing in a `const ComponentInfo` reference may end up being a bit expensive due to the copying of the shared pointers that a `ComponentInfo` holds. One way to avoid this extra expense is to pass in an `ExperimentInfo` object which has access to the `ComponentInfo` object required to make the necessary calls. This method should be cheaper and just as effective. If the classless implementation option is chosen, there would be no need to even store the `m_inst` variable.
```

Comment:
You mean an `ExperimentInfo` `const` `shared_ptr`! We don't want to copy an `ExperimentInfo`

Reply:
Ah yes of course - I'll fix that!